toolkit/tools/pkg/rpmssnapshot/rpmssnapshot.go

```diff
@@ -53,42 +52,38 @@ const (
 	rpmSpecBuiltRPMRegexDistributionIndex
 	rpmSpecBuiltRPMRegexArchitectureIndex
 	rpmSpecBuiltRPMRegexMatchesCount
+
+	chrootName = "rpmssnapshot_chroot"
+	runChecks  = false
 )
 
 type SnapshotGenerator struct {
-	chroot        *safechroot.Chroot
-	buildDirPath  string
-	workerTarPath string
+	simplechroottool.SimpleChrootTool
 }
 
-// New creates an unitialized RPMs snapshot generator.
-func New(buildDirPath, workerTarPath string) *SnapshotGenerator {
-	return &SnapshotGenerator{
-		buildDirPath:  buildDirPath,
-		workerTarPath: workerTarPath,
-	}
+// New creates a new snapshot generator.
+func New(buildDirPath, workerTarPath, specsDirPath, distTag string) (newSnapshotGenerator *SnapshotGenerator, err error) {
+	newSnapshotGenerator = &SnapshotGenerator{}
+	err = newSnapshotGenerator.InitializeChroot(buildDirPath, chrootName, workerTarPath, specsDirPath, distTag, runChecks)
```

Comment:
If we're initializing the chroot here, then we'd need to also expose a `CleanUp` function for the caller. Alternatively, we could still keep the init and clean-up inside `GenerateSnapshot` but I see pros and cons of both without any strong preferences.